src/indexer/index_writer.rs

```diff
@@ -808,7 +808,21 @@ impl<D: Document> IndexWriter<D> {
     /// document queue.
     pub fn add_document(&self, document: D) -> crate::Result<Opstamp> {
         let opstamp = self.stamper.stamp();
-        self.send_add_documents_batch(smallvec![AddOperation { opstamp, document }])?;
+        self.send_add_documents_batch(smallvec![AddOperation {
+            opstamp,
+            document,
+            doc_id: None
+        }])?;
+        Ok(opstamp)
+    }
+
+    pub fn add_document_with_doc_id(&self, doc_id: u32, document: D) -> crate::Result<Opstamp> {
+        let opstamp = self.stamper.stamp();
```

Comment:
should users who call this functions assure that the doc_id must be sorted outside? Or we have assured this, I do not find it.